pandas/core/strings.py

```diff
@@ -1095,24 +1095,61 @@ def str_pad(arr, width, side='left', fillchar=' '):
 
 def str_split(arr, pat=None, n=None):
     """
-    Split each string (a la re.split) in the Series/Index by given
-    pattern, propagating NA values. Equivalent to :meth:`str.split`.
+    Split strings around given separator/delimiter.
+
+    Split each string in the Series/Index by given
```

Comment:
I think generally writing `Series/Index` and `DataFrame/MultiIndex` is not very clear. I'd suggest saying "in the caller's values"